### PR TITLE
Feat form

### PR DESCRIPTION
**Pull request by junaidshaikh-js**
Form component is added, it contains

- simple form fields
- form with validation

Preview: https://deploy-preview-14--hydrogen-ui.netlify.app/docs/form

### CODE REVIEW

File: components/form.css [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+.form-field-validation-message {
+  display: none;
+  position: absolute;
+  bottom: -1rem;
+}
+
+.valid-message {
+  color: rgb(0, 117, 0);
```
Comment (saadahmadhere) [inline components/form.css]: You can use color variables here.

File: components/form.css [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+.invalid-message {
+  color: red;
```
Comment (saadahmadhere) [inline components/form.css]: and here too.

File: docs/form.html [hunk 3]
```diff
@@ -122,7 +122,210 @@ <h5 class="my-1">Components</h5>
...
+                    <div class="row">
+                      <div class="label-container">
+                        <label for="stateList" class="form-label p-md"
+                          >Select Gender</label
+                        >
+                      </div>
+                      <div class="input-container">
+                        <select
+                          class="form-field my-sm p-sm border-sm"
+                          type=""
```
Comment (saadahmadhere) [inline docs/form.html]: type attribute is empty here.

**Review by saadahmadhere — COMMENTED** [review]
Everything else looks good!

File: components/form.css (hunk 1 above)
Comment (junaidshaikh-js) [reply components/form.css]: fixed!

File: components/form.css (hunk 2 above)
Comment (junaidshaikh-js) [reply components/form.css]: fixed

File: docs/form.html (hunk 3 above)
Comment (junaidshaikh-js) [reply docs/form.html]: fixed!

**Review by FarhanMobashir — COMMENTED** [review]
All good ✅

File: components/variable.css [hunk 4]
```diff
@@ -6,6 +6,7 @@
   --white-txt: #fff;
   --black-color: #000;
   --green-color: #006e00;
+  --red-color: #e62727;
```
Comment (shuklahimanshu81) [inline components/variable.css:9]: I will suggest naming as primary, secondary color rather than black or green color

**Review by shuklahimanshu81 — COMMENTED** [review]
Everything else looks good.

**Review by Hardiegogo — COMMENTED** [review]
Looks good.

File: components/variable.css (hunk 4 above)
Comment (junaidshaikh-js) [reply components/variable.css:9]: Primary and secondary colors are already defined, these are literally those colors hence named that way.

Comment (shuklahimanshu81) [reply components/variable.css:9]: Oh, Ok. Maybe I missed.

**Pull request merged**